### DISCUSSION TITLE
"starting folder" parameter for @export_file() / @export_dir

### DISCUSSION THREAD

**Discussion by pixel-boy** [godotengine/godot-proposals#10834] [Editor]
It could be useful to be able to give as parameter a "starting folder" when using @export_file and @export_dir instead of always starting at the "res://" position.

To give a more concrete case, let's imagine a game where all the characters are stored in a "character" folder which itself contains subfolders, I would like to write this:

@export_file(path:"res://character") var character

To avoid having to manually search for the folder when I know that all the files I am interested in are in it, which can become tedious with more complex project organization.

**Comment by kolson25** [comment]
Agree, could see this being really useful.  I come from a web background and in one of the industry standard bundlers "Webpack" there are options to define aliases at a top level for use when importing modules throughout the project. This helps avoid really long absolute paths or a bunch of "../../../" relative paths from every module.

Here's an article that talks about it (more succinct than core docs):  https://www.bitovi.com/blog/how-to-create-a-path-alias-in-webpack 

One note, you may need to put this proposal in the official template format to avoid it being closed. :/

> Only proposals that properly fill out the template will be considered. If the template is not filled out or is filled out improperly, it will be closed.

**Reply by Calinou** [reply]
> One note, you may need to put this proposal in the official template format to avoid it being closed. :/

This only applies to proposal *issues*, not proposal discussions. Discussions are intended to be a more free-form space for coming up with proposal issues when you don't have a precise idea on the implementation.

**Reply by kolson25** [reply]
Ok awesome, thanks for clarifying!  New to Godot contribution so didn't realize the difference there yet.

**Comment by pixel-boy** [comment]
The proposal is not exactly the same but the idea is quite close: https://github.com/godotengine/godot-proposals/issues/9937

**Comment by pixel-boy** [comment]
I'm planning to create a plugin that would allow filtering the quick load menu using an export_custom function that would define a class and a starting folder.
For this to work, I need to be able to manually apply a filter to the quick load menu. Is there a way to do this with a plugin?

I still think this functionality would be useful, especially to compensate for the fact that it's currently not possible to type PackedScene elements, which limits their use as a resource (compared to a custom resource).
